### ISSUE TITLE
Feature request: Multithreading

### ISSUE THREAD

**Issue by JakobAsslaender** [JuliaDiff/FiniteDiff.jl#125]
Hi everyone!
Thanks a ton for implementing this and all associated packages! I use this package wrapped into Optim.jl---by selecting finite difference---and would appreciate improved performance, because who doesn't. So I am wondering if there are any plans to parallelize this code? FD should be embarrassingly parallel, shouldn't it? I would be willing to help, but at this point chances are I would break the code...

Thanks a bunch!

**Comment by ChrisRackauckas** [comment]
Yeah, I think it would be nice if someone added Polyester.jl support to the inner loops here. One would have to be careful with the caching though to make sure it's thread-safe, but that can be done.

**Comment by ChrisRackauckas** [comment]
Someone just needs to do the same thing that was done with https://github.com/JuliaDiff/PolyesterForwardDiff.jl
